frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -461,6 +505,67 @@ class SegmentsView extends React.Component<Props, State> {
     }
   }
 
+  onExpandTree = (expandedKeys: Key[]) => {
+    this.expandGroups(expandedKeys);
+  };
+
+  getSubGroupsAsTreeNodes = (groupId: number) => {
+    if (groupId !== MISSING_GROUP_ID) {
+      return getGroupByIdWithSubgroups(this.props.segmentGroups, groupId)
+        .filter((group) => group !== groupId)
+        .map((group) => getKeyForGroupId(group));
+    }
+    const allSegmentGroups = this.props.segmentGroups.flatMap((group) =>
+      getGroupByIdWithSubgroups(this.props.segmentGroups, group.groupId),
+    );
+    return allSegmentGroups.map((group) => getKeyForGroupId(group));
+  };
+
+  expandGroups = (newExpandedGroups: Key[]) => {
+    if (this.props.visibleSegmentationLayer == null) return;
+    const expandedKeySet = new Set(newExpandedGroups);
+    if (this.state.isRootGroupExpanded && !expandedKeySet.has(getKeyForGroupId(MISSING_GROUP_ID))) {
+      this.setState({ isRootGroupExpanded: false });
+    } else if (
+      !this.state.isRootGroupExpanded &&
+      expandedKeySet.has(getKeyForGroupId(MISSING_GROUP_ID))
+    ) {
+      this.setState({ isRootGroupExpanded: true });
+    }
+    const newGroups = mapGroups(this.props.segmentGroups, (group) => {
+      if (expandedKeySet.has(getKeyForGroupId(group.groupId))) {
+        return {
+          ...group,
+          isExpanded: true,
+        };
+      } else {
+        return {
+          ...group,
+          isExpanded: false,
+          // Close all groups that are not in the expanded list so this method
+          // can be called for every update, e.g. when a group is collapsed.
+        };
+      }
+    });
+    Store.dispatch(setSegmentGroupsAction(newGroups, this.props.visibleSegmentationLayer?.name));
+  };
+
+  collapseGroups = (groupsToCollapse: Key[]) => {
+    if (this.props.visibleSegmentationLayer == null) return;
+    const groupsToCollapseSet = new Set(groupsToCollapse);
+    const newGroups = mapGroups(this.props.segmentGroups, (group) => {
+      if (groupsToCollapseSet.has(getKeyForGroupId(group.groupId))) {
+        return {
+          ...group,
+          isExpanded: false,
+        };
+      } else {
+        return group;
+      }
+    });
```

Comment:
potentially adapt this like suggested for the sibling method. however, this would change the behavior (see my other comment).